internal/provider/oidc_resource.go

```diff
@@ -0,0 +1,260 @@
+package provider
+
+import (
+	"context"
+
+	"github.com/Unleash/unleash-server-api-go/client"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/boolplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/int64planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+)
+
+var (
+	_ resource.Resource              = &oidcResource{}
+	_ resource.ResourceWithConfigure = &oidcResource{}
+)
+
+func NewOidcResource() resource.Resource {
+	return &oidcResource{}
+}
+
+type oidcResource struct {
+	client *client.APIClient
+}
+
+type oidcResourceModel struct {
+	Enabled         types.Bool   `tfsdk:"enabled"`
+	DefaultRootRole types.Int64  `tfsdk:"default_root_role"`
+	DiscoverUrl     types.String `tfsdk:"discover_url"`
+	ClientId        types.String `tfsdk:"client_id"`
+	Secret          types.String `tfsdk:"secret"`
+	AutoCreate      types.Bool   `tfsdk:"auto_create"`
+}
+
+func (r *oidcResource) Configure(ctx context.Context, req resource.ConfigureRequest, _ *resource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	client, ok := req.ProviderData.(*client.APIClient)
+	if !ok {
+		return
+	}
+	r.client = client
+}
+
+func (r *oidcResource) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_oidc"
+}
+
+func (r *oidcResource) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Description: "Manages OIDC configuration.",
+		Attributes: map[string]schema.Attribute{
+			"enabled": schema.BoolAttribute{
+				Description: "Whether or not OIDC is enabled.",
+				Required:    true,
+			},
+			"discover_url": schema.StringAttribute{
+				Description: "A URL pointing to the .well-known configuration of the OIDC provider.",
+				Required:    true,
+			},
+			"client_id": schema.StringAttribute{
+				Description: "The OIDC public identifier.",
+				Required:    true,
+			},
+			"secret": schema.StringAttribute{
+				Description: "The OIDC secret.",
+				Required:    true,
+			},
+			"auto_create": schema.BoolAttribute{
+				Description: "Whether to auto create users when they login to Unleash for the first time.",
+				Optional:    true,
+				Computed:    true,
+				PlanModifiers: []planmodifier.Bool{
```

Comment:
This lets teraform assume the last state it got will be unchanged and reuse that state.This is necessary for optional properties here because they cannot be removed on the Unleash side and will confuse the snot out of terraform